### PR TITLE
Cleanup material/tabs.dart documentation

### LINKED ISSUE #121523: tabs.dart documentation needs some cleanup
The documentation in `material/tabs.dart` needs some cleanup. For instance:
- “See also” section of `TabBar` is referencing itself.
- `indicatorPadding` related documentation needs improvements related to some logic changes added in https://github.com/flutter/flutter/pull/63683.

### PR DESCRIPTION
**Pull request by bleroux**
## Description

The documentation in `material/tabs.dart` needs some cleanup. For instance:
- “See also” section of `TabBar` is referencing itself.
- `indicatorPadding` related documentation needs improvements related to some logic changes added in https://github.com/flutter/flutter/pull/63683.

## Related Issue

Fixes https://github.com/flutter/flutter/issues/121523

## Tests

Documentation only.

### CODE REVIEW

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -573,11 +573,10 @@ class _TabBarScrollPosition extends ScrollPositionWithSingleContext {
...
     // to the initial scroll position to avoid strange scrolling effects in
-    // release mode: In release mode, the viewport temporarily may have a
-    // dimension of zero before the actual dimension is calculated. In that
-    // scenario, setting the actual dimension would cause a strange scroll
-    // effect without this guard because the super call below would starts a
-    // ballistic scroll activity.
+    // release mode: the viewport temporarily may have a dimension of zero
+    // before the actual dimension is calculated. In that scenario, setting
+    // the actual dimension would cause a strange scroll effect without this
+    // guard because the super call below would starts a ballistic scroll activity.
```
Comment (navaronbracke) [inline packages/flutter/lib/src/material/tabs.dart:579]: While you're fixing up the docs here, `would starts a ballistic scroll activity` should be `would start a ballistic scroll activity`.

Comment (bleroux) [reply packages/flutter/lib/src/material/tabs.dart:579]: @navaronbracke  Thanks! I will fix this too.

**Comment by Hixie** [comment]
test-exempt: documentation

These fixes are awesome.

**Review by Hixie — APPROVED** [review]
LGTM

**Pull request merged**